pkg/kstatus/observe/observers/common.go

```diff
@@ -0,0 +1,171 @@
+package observers
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sort"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/cli-utils/pkg/kstatus/observe/event"
+	"sigs.k8s.io/cli-utils/pkg/kstatus/observe/observer"
+	"sigs.k8s.io/cli-utils/pkg/kstatus/status"
+	"sigs.k8s.io/cli-utils/pkg/kstatus/wait"
+)
+
+// BaseObserver provides some basic functionality needed by the observers.
+type BaseObserver struct {
+	Reader observer.ClusterReader
+
+	Mapper meta.RESTMapper
+
+	computeStatusFunc observer.ComputeStatusFunc
+}
+
+// SetComputeStatusFunc allows for setting the function used by the observer for computing status. The default
+// value here is to use the status package. This is provided for testing purposes.
+func (b *BaseObserver) SetComputeStatusFunc(statusFunc observer.ComputeStatusFunc) {
+	b.computeStatusFunc = statusFunc
+}
+
+// LookupResource looks up a resource with the given identifier. It will use the rest mapper to resolve
+// the version of the GroupKind given in the identifier.
+// If the resource is found, it will be returned and the observedResource will be nil. If there is an error
```

Comment:
This comment doesn't match the function's return values anymore

Reply:
Good catch. I have created a PR to fix it.